optimizely/optimizely.py

```diff
@@ -316,7 +320,8 @@ def track(self, event_key, user_id, attributes=None, event_tags=None):
       except:
         error = sys.exc_info()[1]
         self.logger.log(enums.LogLevels.ERROR, 'Unable to dispatch conversion event. Error: %s' % str(error))
-
+      self.notification_center.fire_notifications(enums.NotificationTypes.TRACK, event_key, user_id,
```

Comment:
This actually is interesting. Do we fire the notification as soon as we dispatch the event, or we wait for the logendpoint to return to us before we tell the listener? I could see us running into counting issues (mismatches) between us and other analytics providers if we pass the event without first guaranteeing that we received it on our end. Thoughts @aliabbasrizvi @wangjoshuah @ceimaj ?

Reply:
IMHO, I think if there is a mismatch then there would be a problem on our end and it would probably bubble up faster.

Reply:
Yeah I don't feel too strongly about it. The key here is to very clearly document in our dev docs, WHEN exactly we fire off the notification.

Reply:
well, that actually brings up a good point, do we want to have pre and post calls?

Reply:
nah, I think that gets too complicated. The current approach you have there should be good.